ui-admin/src/study/families/FamilyStructureTable.tsx

```diff
@@ -0,0 +1,261 @@
+
+
+  const columns: ColumnDef<FamilyMemberWithSubrows>[] = [
+    {
+      id: 'expanded',
+      header: '',
+      accessorKey: 'expanded',
+      cell: ({ row }) => {
+        return <div
+          style={{
+            paddingLeft: `${row.depth * 2}rem`
+          }}
+        >
+          <button
+            className="btn btn-link"
+            onClick={() => row.toggleExpanded()}
+            disabled={!row.getCanExpand()}
+          >
+            {row.getIsExpanded() ? <FontAwesomeIcon icon={faCaretUp}/> : <FontAwesomeIcon icon={faCaretDown}/>}
+          </button>
+        </div>
+      }
+    },
+    {
+      id: 'shortcode',
+      header: 'Shortcode',
+      accessorKey: 'member.shortcode',
+      cell: ({ row }) => {
+        return <Link to={`${studyEnvContext.currentEnvPath}/participants/${row.original.member.shortcode}`}>
+          {row.original.member.shortcode}
+        </Link>
+      }
+    },
+    {
+      id: 'name',
+      header: 'Name',
+      accessorKey: 'member.profile.givenName',
+      cell: ({ row }) => {
+        return `${row.original.member.profile?.givenName || ''} ${row.original.member.profile?.familyName || ''}`.trim()
+      }
+    },
+    {
+      id: 'proxyRelation',
+      header: 'Proxies',
+      cell: ({ row }) => {
+        return <ProxyList
+          member={row.original.member}
+          family={family}
+          studyEnvContext={studyEnvContext}/>
+      }
+    }
+  ]
+
+  const table = useReactTable({
+    data: members,
+    columns,
+    state: {
+      expanded,
+      sorting
+    },
+    onExpandedChange: setExpanded,
+    enableExpanding: true,
+    getSubRows: row => row.subrows,
+
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getExpandedRowModel: getExpandedRowModel()
+  })
+
+  const customRowHeader = (row: Row<FamilyMemberWithSubrows>) => {
+    const parentRow = row.getParentRow()
+    if (!isNil(parentRow)) {
+      const parent = parentRow.original
+      const relation = row.original.relationToParentRow?.familyRelationship
+      const subRowIdx = parent.subrows.findIndex(s => s.member.id === row.original.member.id)
+      const showHeader = (
+        subRowIdx === 0 || parent.subrows[subRowIdx - 1].relationToParentRow?.familyRelationship != relation
+      )
+      if (showHeader) {
+        return <tr>
+          <td colSpan={table.getFlatHeaders().length}>
+            <div style={{
+              paddingLeft: `${row.depth * 2 }rem`
+            }}>
+              <span className='fst-italic'>
+                {startCase(lowerCase(relation))} of:
+              </span>
+            </div>
+          </td>
+        </tr>
+      }
+    }
+    return <></>
+  }
+
+  useEffect(() => {
+    table.toggleAllRowsExpanded(true)
+  }, [])
+
+  return <div>
+    {basicTableLayout(table, {
+      customRowHeader,
+      tableClass: 'table table-striped m-0'
+    })}
```

Comment:
sorting a column causes the relationship groupings to get incorrectly shuffled around:

![Screenshot 2024-06-25 at 1 32 36 PM](https://github.com/broadinstitute/juniper/assets/7257391/371c393e-f742-4b1d-9170-6aa6fd8098d6)

I assume that's not going to be a super trivial fix, and I wonder if it makes sense to just disable sorting for this table anyway?